first_seminar/bank/src/test/java/org/example/domain/accountTest/InstallmentSavingAccountTest.java

```diff
@@ -0,0 +1,82 @@
+package org.example.domain.accountTest;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+import org.example.domain.account.InstallmentSavingAccount;
+import org.example.domain.account.SavingAccount;
+import org.example.domain.account.constant.AccountStatus;
+import org.example.domain.product.InstallmentSavingProduct;
+import org.example.domain.product.SavingProduct;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class InstallmentSavingAccountTest {
+
+    InstallmentSavingProduct militarySaving;
+    InstallmentSavingAccount dundeoAccount;
+
+    @BeforeEach
+    void init() {
+        militarySaving = InstallmentSavingProduct.getInstance(0);
+        dundeoAccount = new InstallmentSavingAccount("0000", "덩둔더", 100_000L, militarySaving);
+    }
+
+    @Test
+    void 십만원을_입금하면_잔액이_십만원_늘어난다() {
+        // given
+        Long expected = dundeoAccount.getBalance() + 10_000L;
+        // when
+        dundeoAccount.deposit(10_000L);
+        Long actual = dundeoAccount.getBalance();
+
+        // then
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    @Test
+    void 해지를_시도했는데_만기되지_않았다면_해지는_되지않고_만기되지_않았음을_반환한다() {
+        // given
+        SavingProduct studentSaving = SavingProduct.getInstance(0);
+        SavingAccount junseoAccount = new SavingAccount("0000", "정준서", 10_000L, studentSaving);
+        AccountStatus expectedStatus = AccountStatus.NOT_EXPIRED;
+        Long expectedBalance = junseoAccount.getBalance();
+
+        // when
+        AccountStatus actualStatus = dundeoAccount.closure(junseoAccount);
+        Long actualBalance = junseoAccount.getBalance();
+
+        // then
+        assertAll(
+            () -> assertThat(actualStatus).isEqualTo(expectedStatus),
+            () -> assertThat(actualBalance).isEqualTo(expectedBalance)
+        );
```

Comment:
```suggestion
    assertThat(actualStatus).isEqualTo(expectedStatus),
    assertThat(actualBalance).isEqualTo(expectedBalance)
```
제가 TDD 공부한지 얼마 안되서 그러는데 혹시 assertAll을 사용한 것과 사용 안 한 것에 차이가 있나요?!!?

Reply:
assertThat만 여러 개 사용하게 되면 assertThat 하나를 실패했을 때 그 뒤에 있는 assertThat은 실행하지 않아요!
여러 개의 asserThat을 assertAll로 묶어야 전체 중간에 실패가 발생하더라도 전체 asserThat을 실행하기 때문에
전체 assertThat들 중에서 어떤 것들이 실패했는지 알 수 있답니당

Reply:
https://zzang9ha.tistory.com/418
여기 보시면 바로 이해되실 거 같아여